model/src/lib.rs

```diff
@@ -51,35 +52,94 @@ pub struct UserOrder {
     pub signature: Signature,
 }
 
-impl UserOrder {
+impl OrderCreation {
     pub fn token_pair(&self) -> Option<TokenPair> {
         TokenPair::new(self.buy_token, self.sell_token)
     }
 }
 
-/// An order that is returned when querying the orderbook.
-///
-/// Contains extra fields thats are populated by the orderbook.
-#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize)]
+// uid as 56 bytes: 32 for orderDigest, 20 for ownerAddress and 4 for validTo
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct OrderUid(pub [u8; 56]);
+
+impl Default for OrderUid {
```

Comment:
nit: Do we need Default on OrderUid? Maybe we can just implement default on UserOrder which could create the correct ID on an empty order. It could be weird to use an id that doesn't confirm to the derived format.